### PR TITLE
[runtime] Implement SwiftObject -methodForSelector: and +instanceMethodForSelector:

### PR DESCRIPTION
**Pull request by gparker42**
Fixes SR-5577 and rdar://33606034.

### CODE REVIEW

File: stdlib/public/runtime/SwiftObject.h [hunk 1]
```diff
@@ -58,6 +58,9 @@ SWIFT_RUNTIME_EXPORT @interface SwiftObject<NSObject> {
 - (BOOL)conformsToProtocol:(Protocol *)aProtocol;
 
 - (BOOL)respondsToSelector:(SEL)aSelector;
++ (BOOL)instancesRespondToSelector:(SEL)aSelector;
```
Comment (gparker42) [inline stdlib/public/runtime/SwiftObject.h]: +instancesRespondToSelector: was already implemented, but it was not declared here.

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test linux platform

**Comment by gparker42** [comment]
@swift-ci please test linux platform

**Comment by DougGregor** [comment]
Those Linux failures are completely unrelated. It looks like dispatch didn't get fully rebuilt

**Comment by gparker42** [comment]
@swift-ci please test linux platform

**Pull request merged**